infrastructure/quick-deploy/aws/storage.tf

```
@@ -412,4 +419,54 @@ locals {
     use_chunk_encoding    = true
     use_check_sum         = true
   }
+
+  mongodb_pvc_provisioner = try(
+    coalesce(var.mongodb_sharding.persistence.shards.storage_provisioner),
+    coalesce(var.mongodb.persistent_volume.storage_provisioner),
+    coalesce(var.mongodb_ebs.mongodb != null ? "ebs.csi.aws.com" : null),
+    coalesce(var.mongodb_efs.mongodb != null ? "efs.csi.aws.com" : null),
+    ""
+  )
+
+  mongodb_configsvr_pvc_provisioner = try(
+    coalesce(var.mongodb_sharding.persistence.configsvr.storage_provisioner),
+    coalesce(var.mongodb_ebs.configsvr != null ? "ebs.csi.aws.com" : null),
+    coalesce(var.mongodb_efs.configsvr != null ? "efs.csi.aws.com" : null),
+  "")
+
+  # Ensures some mandatory storage class parameters are effectively passed when persistence is enabled
+  mongodb_storage_class_parameters = merge(
+    local.mongodb_pvc_provisioner == "efs.csi.aws.com" ? {
+      provisioningMode = "efs-ap"
+      fileSystemId     = module.mongodb_efs_persistent_volume[0].id
+      directoryPerms   = "755"
+      uid              = var.mongodb.security_context.run_as_user # optional
+      gid              = var.mongodb.security_context.fs_group    # optional
+      basePath         = "/mongodb"                               # optional
+    } : local.mongodb_pvc_provisioner == "ebs.csi.aws.com" ?
```

You can avoid ternary chaining by leveraging merge:

```suggestion
    } : null,
    local.mongodb_pvc_provisioner == "ebs.csi.aws.com" ?
```